src/Chapter1.hs

```diff
@@ -541,8 +542,12 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x <= y && z >= x = if z <= y then z else y
+    | x >= y && z >= x = x
+    | x >= y && z <= x = if z <= y then y else z
+    | x <= y && z <= x = x
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = `, to tell the compiler, that your pattern matching is exhaustive 🙂
In this case, you correctly noticed that all these cases covering all possibilities. So we know exactly, that if we get to the last case, and all previous didn't fit, then this one will always work. So you can return `x` without checking these conditions:
```suggestion
    | otherwise = x
```